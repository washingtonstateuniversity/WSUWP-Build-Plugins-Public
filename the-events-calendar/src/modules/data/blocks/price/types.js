/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_STORE } from '@moderntribe/events/data/utils';

export const SET_INITIAL_STATE = `${ PREFIX_EVENTS_STORE }/PRICE_BLOCK/SET_INITIAL_STATE`;

export const SET_PRICE_COST = `${ PREFIX_EVENTS_STORE }/SET_PRICE_COST`;

export const SET_PRICE_SYMBOL = `${ PREFIX_EVENTS_STORE }/SET_PRICE_SYMBOL`;

export const SET_PRICE_POSITION = `${ PREFIX_EVENTS_STORE }/SET_PRICE_POSITION`;

export const SET_PRICE_DESCRIPTION = `${ PREFIX_EVENTS_STORE }/SET_PRICE_DESCRIPTION`;
