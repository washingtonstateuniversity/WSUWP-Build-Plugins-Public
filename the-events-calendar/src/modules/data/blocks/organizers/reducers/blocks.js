/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import * as types from './../types';
import block from './block';

export const byId = ( state = {}, action ) => {
	switch ( action.type ) {
		case types.ADD_ORGANIZER_BLOCK:
		case types.REMOVE_ORGANIZER_BLOCK:
			return {
				...state,
				[ action.payload.id ]: block( state[ action.payload.id ], action ),
			};
		default:
			return state;
	}
};

export const allIds = ( state = [], action ) => {
	switch ( action.type ) {
		case types.ADD_ORGANIZER_BLOCK:
			return [ ...state, action.payload.organizer ];
		case types.REMOVE_ORGANIZER_BLOCK:
			return [ ...state ].filter( ( organizer ) => organizer !== action.payload.organizer );
		default:
			return state;
	}
};

export default combineReducers( {
	byId,
	allIds,
} );
