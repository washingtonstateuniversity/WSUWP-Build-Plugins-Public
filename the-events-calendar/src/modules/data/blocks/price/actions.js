/**
 * Internal dependencies
 */
import * as types from './types';

export const setInitialState = ( props ) => ( {
	type: types.SET_INITIAL_STATE,
	payload: props,
} );

export const setCost = ( cost ) => ( {
	type: types.SET_PRICE_COST,
	payload: {
		cost,
	},
} );

export const setPosition = ( position ) => ( {
	type: types.SET_PRICE_POSITION,
	payload: {
		position,
	},
} );

export const togglePosition = ( showBefore ) => {
	return setPosition( showBefore ? 'prefix' : 'suffix' );
};

export const setSymbol = ( symbol ) => ( {
	type: types.SET_PRICE_SYMBOL,
	payload: {
		symbol,
	},
} );

export const setDescription = ( description ) => ( {
	type: types.SET_PRICE_DESCRIPTION,
	payload: {
		description,
	},
} );
