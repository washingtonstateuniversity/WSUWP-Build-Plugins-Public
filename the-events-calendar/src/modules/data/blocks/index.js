/**
 * Internal dependencies
 */
import * as datetime from './datetime';

export { datetime };
export { default } from './reducer';
