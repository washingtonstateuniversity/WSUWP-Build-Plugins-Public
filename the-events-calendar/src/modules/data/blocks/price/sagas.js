/**
 * External Dependencies
 */
import { call, put, all, takeEvery } from 'redux-saga/effects';

/**
 * WordPress dependencies
 */
import { select } from '@wordpress/data';

/**
 * Internal dependencies
 */
import * as types from './types';
import { DEFAULT_STATE } from './reducer';
import * as actions from './actions';
import { priceSettings } from '@moderntribe/common/utils/globals';

export function* setInitialState( action ) {
	const { get } = action.payload;
	const settings = yield call( priceSettings );
	const isNewEvent = yield call( [ select( 'core/editor' ), 'isEditedPostNew' ] );

	const currencySymbol = isNewEvent
		? settings.defaultCurrencySymbol
		: get( 'currencySymbol', DEFAULT_STATE.symbol );

	const currencyPosition = isNewEvent
		? settings.defaultCurrencyPosition
		: get( 'currencyPosition', DEFAULT_STATE.position );

	yield all( [
		put( actions.setPosition( currencyPosition ) ),
		put( actions.setSymbol( currencySymbol ) ),
		put( actions.setCost( get( 'cost', DEFAULT_STATE.cost ) ) ),
		put( actions.setDescription( get( 'costDescription', DEFAULT_STATE.description ) ) ),
	] );
}

export default function* watchers() {
	yield takeEvery( types.SET_INITIAL_STATE, setInitialState );
}
