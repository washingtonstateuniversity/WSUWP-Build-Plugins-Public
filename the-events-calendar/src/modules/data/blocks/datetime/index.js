/**
 * Internal dependencies
 */
import reducer from './reducer';
import * as types from './types';
import * as actions from './actions';
import * as thunks from './thunks';
import * as selectors from './selectors';
import sagas from './sagas';

export default reducer;
export { types, actions, thunks, selectors, sagas };
