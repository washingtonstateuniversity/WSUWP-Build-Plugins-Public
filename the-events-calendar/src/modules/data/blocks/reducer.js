/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import datetime from './datetime';
import organizers from './organizers';
import price from './price';
import website from './website';
import venue from './venue';
import classic from './classic';
import sharing from './sharing';

export default combineReducers( {
	datetime,
	classic,
	venue,
	organizers,
	price,
	website,
	sharing,
} );
