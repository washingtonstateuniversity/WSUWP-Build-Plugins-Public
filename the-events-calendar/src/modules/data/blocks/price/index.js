/**
 * Internal dependencies
 */
import reducer from './reducer';

import * as selectors from './selectors';
import * as actions from './actions';
import * as types from './types';
import sagas from './sagas';

export default reducer;
export { selectors, actions, types, sagas };
