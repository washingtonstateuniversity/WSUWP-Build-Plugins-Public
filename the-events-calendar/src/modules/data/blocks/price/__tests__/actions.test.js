/**
 * External dependencies
 */
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
/**
 * Internal dependencies
 */
import { actions } from '@moderntribe/events/data/blocks/price';

const middlewares = [ thunk ];
const mockStore = configureStore( middlewares );

describe( '[STORE] - Price actions', () => {
	it( 'Should set initial state', () => {
		expect( actions.setInitialState( {} ) ).toMatchSnapshot();
	} );
	it( 'Should set the cost', () => {
		expect( actions.setCost( 10 ) ).toMatchSnapshot();
	} );

	it( 'Should set the position of the symbol', () => {
		expect( actions.setPosition( 'suffix' ) ).toMatchSnapshot();
	} );

	it( 'Should set the price description', () => {
		expect( actions.setDescription( 'My description' ) ).toMatchSnapshot();
	} );

	it( 'Should set the symbol', () => {
		expect( actions.setSymbol( 'â‚¬' ) ).toMatchSnapshot();
	} );

	it( 'Should toggle the prefix', () => {
		expect( actions.togglePosition( true ) ).toMatchSnapshot();
		expect( actions.togglePosition( false ) ).toMatchSnapshot();
	} );
} );
