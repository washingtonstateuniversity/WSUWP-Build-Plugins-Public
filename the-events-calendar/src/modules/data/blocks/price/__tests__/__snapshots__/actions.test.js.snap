// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - Price actions Should set initial state 1`] = `
Object {
  "payload": Object {},
  "type": "@@MT/EVENTS/PRICE_BLOCK/SET_INITIAL_STATE",
}
`;

exports[`[STORE] - Price actions Should set the cost 1`] = `
Object {
  "payload": Object {
    "cost": 10,
  },
  "type": "@@MT/EVENTS/SET_PRICE_COST",
}
`;

exports[`[STORE] - Price actions Should set the position of the symbol 1`] = `
Object {
  "payload": Object {
    "position": "suffix",
  },
  "type": "@@MT/EVENTS/SET_PRICE_POSITION",
}
`;

exports[`[STORE] - Price actions Should set the price description 1`] = `
Object {
  "payload": Object {
    "description": "My description",
  },
  "type": "@@MT/EVENTS/SET_PRICE_DESCRIPTION",
}
`;

exports[`[STORE] - Price actions Should set the symbol 1`] = `
Object {
  "payload": Object {
    "symbol": "â‚¬",
  },
  "type": "@@MT/EVENTS/SET_PRICE_SYMBOL",
}
`;

exports[`[STORE] - Price actions Should toggle the prefix 1`] = `
Object {
  "payload": Object {
    "position": "prefix",
  },
  "type": "@@MT/EVENTS/SET_PRICE_POSITION",
}
`;

exports[`[STORE] - Price actions Should toggle the prefix 2`] = `
Object {
  "payload": Object {
    "position": "suffix",
  },
  "type": "@@MT/EVENTS/SET_PRICE_POSITION",
}
`;
